src/commands/PublishCommand.js

```diff
@@ -542,6 +542,8 @@ export default class PublishCommand extends Command {
 
     this.updates.forEach((update) => {
       this.execScript(update.package, "prepublish");
+      this.execScript(update.package, "prepare");
```

Comment:
As I understand it, we want to ensure that a given set of lifecycle scripts finishes in a package before any dependent packages start their prepublish+ lifecycles. So `PackageUtilities.runParallelBatches` is definitely appropriate here. I'm not the sharpest with the async loops, but what you describe sounds like the right approach.

In any case, lots of testing!